go/arrow/memory/buffer.go

```diff
@@ -60,15 +61,28 @@ func (b *Buffer) Release() {
 	}
 }
 
+// Reset resets the buffer for reuse.
+func (b *Buffer) Reset(buf []byte) {
```

Comment:
it's a part of the codebase I am not completely fluent with.
but, shouldn't we also somewhat test whether that `Buffer` is `Mutable()` before modifying it?

Reply:
As far as I can tell `mutable` just means `there is a reference to an allocator here so things can be resized`, I don't think it has much bearing on calling `Reset()`.

The `Reset()` is used to reuse data structures so you're also kind of already saying: "I know what I'm doing"

Reply:
ok